### PR TITLE
chapter_12

### PR DESCRIPTION
**Pull request by lrmi0808**
Ready

### CODE REVIEW

**Comment by lrmi0808** [comment]
Ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
```
Comment (huongnt-2545) [inline app/assets/stylesheets/password_resets.scss:1]: delete this file, it's not useful

File: app/helpers/password_resets_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module PasswordResetsHelper
```
Comment (huongnt-2545) [inline app/helpers/password_resets_helper.rb:1]: delete this file

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
```
Comment (huongnt-2545) [inline config/routes.rb:2]: move line 2, 3 into scope at line 4

File: test/controllers/password_resets_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/password_resets_controller_test.rb:1]: delete this file

**Comment by lrmi0808** [comment]
Ready

File: app/controllers/password_resets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:4]: combine into 1 line please

File: app/controllers/password_resets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  def new
+  end
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:7]:
```suggestion
  def new; end
```

File: app/controllers/password_resets_controller.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      flash[:info] = t("email_reset_sent")
+      redirect_to home_url
+    else
+      flash.now[:danger] = t("email_not_found")
+      render :new
+    end
+  end
+
+  def edit
+  end
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:23]:
```suggestion
  def edit; end
```

File: config/routes.rb [hunk 8]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
    scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
 
+      get 'password_resets/new'
+      get 'password_resets/edit'
```
Comment (anlvq-1058) [inline config/routes.rb:5]: Do you need to keep  line 4 and 5

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by lrmi0808** [comment]
Ready

File: config/routes.rb [hunk 9]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
    scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
 
+      # get 'password_resets/new'
+      # get 'password_resets/edit'
```
Comment (anlvq-1058) [inline config/routes.rb:5]: remove lines 4, 5 if you don't use it

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by lrmi0808** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**